server/src/main/java/org/opensearch/identity/tokens/BasicAuthToken.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.tokens;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+/**
+ * Basic (Base64 encoded) Authentication Token in a http request header
+ */
+public final class BasicAuthToken implements AuthToken {
+
+    public final static String TOKEN_IDENIFIER = "Basic";
+
+    private String user;
+    private String password;
+
+    public BasicAuthToken(final String headerValue) {
+        final String base64Encoded = headerValue.substring(TOKEN_IDENIFIER.length()).trim();
+        final byte[] rawDecoded = Base64.getDecoder().decode(base64Encoded);
+        final String usernamepassword = new String(rawDecoded, StandardCharsets.UTF_8);
+
+        final String[] tokenParts = usernamepassword.split(":", 2);
+        if (tokenParts.length != 2) {
+            // TODO: Wrong except type
+            throw new IllegalStateException("Illegally formed basic authorization header " + tokenParts[0]);
+        }
+        user = tokenParts[0];
+        password = tokenParts[1];
+    }
+
+    public String getUser() {
+        return user;
+    }
+
+    // TODO: Password should not be actually stored this way, but lets handle this in a future PR
+    public String getPassword() {
+        return user;
+    }
```

Comment:
we can delete this method if not needed

Reply:
We need this method, but I'll clean up the comments around it.

Reply:
should we return `user` here?

Reply:
Yes, this was updated - check the latest version